packages/app/src/app/overmind/internalActions.ts

```diff
@@ -163,33 +161,44 @@ export const authorize = async ({ state, effects }: Context) => {
 };
 export const runProviderAuth = (
   { effects, state }: Context,
-  {
-    provider,
-    useExtraScopes,
-  }: { useExtraScopes?: boolean; provider?: 'apple' | 'github' | 'google' }
+  options: AuthOptions
 ) => {
-  const hasDevAuth = process.env.LOCAL_SERVER || process.env.STAGING;
+  const { provider } = options;
+
+  // When in development, check if there's authentication.
+  const isInitialDevelopmentAuth =
+    process.env.NODE_ENV === 'development' && !state.hasLogIn;
+  // Use dev auth if local server or staging and there's no
+  // authentication or the provider isn't GitHub. This is
+  // needed to allow us to go to GitHub to authorize extra
+  // scopes during development.
+  const useDevAuth =
+    (process.env.LOCAL_SERVER || process.env.STAGING) &&
+    (isInitialDevelopmentAuth || provider !== 'github');
+  // Base path
+  const baseUrl = useDevAuth
+    ? location.origin
+    : process.env.ENDPOINT || 'https://codesandbox.io';
 
   const authPath = new URL(
-    location.origin + (hasDevAuth ? '/auth/dev' : `/auth/${provider}`)
+    baseUrl + (useDevAuth ? '/auth/dev' : `/auth/${provider}`)
   );
 
   authPath.searchParams.set('version', '2');
 
   if (provider === 'github') {
-    if (useExtraScopes) {
-      authPath.searchParams.set('scope', 'user:email,repo,workflow');
-    } else {
-      authPath.searchParams.set('scope', 'user:email');
-    }
+    authPath.searchParams.set(
+      'scope',
+      'useExtraScopes' in options ? 'user:email,repo,workflow' : 'user:email'
```

Comment:
I believe that when `useExtraScopes` is explicitly set to `undefined` this also results in `true` when using `in`:

```js
const a = { b: undefined };
console.log('b' in a); // true
```
I think you changed it so `useExtraScopes` is always omitted (`signInButtonClicked({ provider });`) when we don't need it, but there might be cases where it's not omitted and might be undefined. Should we change this to `options.useExtraScopes` instead to be sure? Wdyt?

Reply:
@tristandubbeld nice catch! I fixed it here: [2ab2c57](https://github.com/codesandbox/codesandbox-client/pull/7388/commits/2ab2c57a6a7c80dd726cbf8bd9066c0ee99fe893)